flowmachine/flowmachine/core/server/query_schemas/aggregation_unit.py

```diff
@@ -0,0 +1,33 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Definition of a custom marshmallow field for aggregation units, and function
+for getting the corresponding SpatialUnit object.
+"""
+
+from flowmachine.core import make_spatial_unit
+
+from marshmallow.fields import String
+from marshmallow.validate import OneOf
+
+
+class AggregationUnit(String):
```

Comment:
(Maybe not now, but) this is a prime target for shifting to a JSON object which maps to the spatial objects.

Reply:
Yes, I agree. I haven't attempted to implement this shift, partly because I've attempted to avoid changing any behaviour in this PR and partly because I think this change would be best as part of a wider consideration of how we want to expose aggregation units (and the associated permissions). But anticipation of this change was one of the reasons for moving this definition out of `custom_fields.py`.